catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/AccessibilityPreferencesActivityTest.java

```diff
@@ -0,0 +1,156 @@
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_ADDITIONAL_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_DRAGNDROP_DELAY;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_FONTFACE_VALUE_DYSLEXIC;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_HIGH_CONTRAST;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_HIGH_CONTRAST_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_LARGE_ELEMENT_SPACING;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_LARGE_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_LARGE_TEXT;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_ADDITIONAL_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_DRAGNDROP_DELAY;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_FONTFACE;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_HIGH_CONTRAST;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_HIGH_CONTRAST_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_LARGE_ELEMENT_SPACING;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_LARGE_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_LARGE_TEXT;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_STARTER_BRICKS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_STARTER_BRICKS;
+
+@RunWith(AndroidJUnit4.class)
+public class AccessibilityPreferencesActivityTest {
+
+	private List<String> settings = new ArrayList<>(Arrays.asList(ACCESS_LARGE_TEXT, ACCESS_HIGH_CONTRAST,
+			ACCESS_ADDITIONAL_ICONS, ACCESS_LARGE_ICONS, ACCESS_HIGH_CONTRAST_ICONS, ACCESS_LARGE_ELEMENT_SPACING,
+			ACCESS_STARTER_BRICKS, ACCESS_DRAGNDROP_DELAY));
+
+	@Rule
+	public BaseActivityInstrumentationRule<AccessibilityPreferencesActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(AccessibilityPreferencesActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@After
+	public void tearDown() {
+	}
+
+	@Test
+	public void settingsTest() {
+		SharedPreferences sharedPreferences = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		onView(withText(R.string.preference_access_title_fontface)).perform(click());
+		onView(withText(R.string.preference_access_title_fontface_dyslexic)).perform(click());
+		assertEquals(sharedPreferences.getString(ACCESS_MYPROFILE_FONTFACE, ""), ACCESS_FONTFACE_VALUE_DYSLEXIC);
+
+		onView(withText(R.string.close)).perform(click());
+
+		onView(withText(R.string.preference_access_title_large_text)).perform(scrollTo(), click());
```

Comment:
if youd use onData you could skip on the scrollTo() part, which in my experience is not always doing what you want. (at least check this test on a tablet and a very small phone)

Reply:
So I tried using onData with a custom Matcher instead, and racked my brain why it didn't work. It would check the first three (visible) checkboxes correctly, and then do absolutely nothing. The test just kept running indefinitely.
Turns out it doesn't work because the AdapterView is inside a ScrollVIew, and from what I've gathered on the internet, a ScrollView should be used in conjunction with the scrollTo() action.
Furthermore the AdapterView is completely collapsed, meaning that its bounding box is big enough to display all of its elements, and the AdapterView itself can't be scrolled anymore. This is necessary for the ScrollView to work as it does.

So this is a bit of a special case, and I don't think using scrollTo() can be avoided. What do you think?